### ISSUE TITLE
Exception due to missing `ThemeTogglerService` when disabling the Themes module

### ISSUE THREAD

**Issue by RitaBon** [OrchardCMS/OrchardCore#12707]
Hello, I was playing around with the features switches and by disabling one of them (not entirely sure which one but I think it was one around Themes), this error started popping up:

InvalidOperationException: No service for type 'OrchardCore.Themes.Services.DarkModeService' has been registered.

Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
Microsoft.AspNetCore.Mvc.Razor.RazorPagePropertyActivator+<>c__DisplayClass8_0.<CreateActivateInfo>b__1(ViewContext context)
Microsoft.Extensions.Internal.PropertyActivator<TContext>.Activate(object instance, TContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorPagePropertyActivator.Activate(object page, ViewContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorPageActivator.Activate(IRazorPage page, ViewContext context)
Microsoft.AspNetCore.Mvc.Razor.RazorView.RenderPageCoreAsync(IRazorPage page, ViewContext context)

I can't access any of the pages or login/logoff, so not entirely sure how to fix this. After searching for a resolution for some time, haven't been able to find this mentioned anywhere.

**Comment by hishamco** [comment]
This SHOULD happen if `OC.Themes` module is disabled

**Comment by hishamco** [comment]
The fix should be easy, I will push a PR soon

**Comment by RitaBon** [comment]
Thank you!

**Comment by hishamco** [comment]
I didn't try it before, but I'm sure @Skrypt already did ;)

**Comment by Skrypt** [comment]
That, or remove the service injection in the Layout Razor templates in a custom theme if you don't want to have Theming enabled.

**Comment by hishamco** [comment]
But it's there in the default admin theme

**Comment by Skrypt** [comment]
Yes, because we need it. But if someone doesn't then it is injected in a Razor view so it can be disposed of. I don't think we should make the Theme feature always enabled because some people are using OC as a Framework without the Themes feature.

**Comment by hishamco** [comment]
> Yes, because we need it

Of course, we do

>  I don't think we should make the Theme feature always enabled because some people are using OC as a Framework without the Themes feature.

It's not mendantory, but once it's enable we can't disable it

**Comment by jtkech** [comment]
Maybe just make `TheAdmin` (or in the meantime a currently selected custom admin theme) dependent on `OC.Themes` in its manifest.

To unblock the situation, try the following from the application

    ..AddOrchardCms()
    .ConfigureServices(tenantServices =>
        tenantServices.AddScoped<DarkModeService, DarkModeService>())
    .AddSetupFeatures("OrchardCore.AutoSetup");

**Comment by RitaBon** [comment]
Thank you all for your help! Adding scoped DarkModeService worked like a charm. I don't really need the themes since I'm trying to use Orchard as a headless CMS and enabling themes seems to mess with the preview of widgets.

**Comment by hishamco** [comment]
I will submit another PR to resolve this issue by adding a default implementation to avoid crash the app
